packages/flutter/lib/src/widgets/image_icon.dart

```diff
@@ -76,22 +83,29 @@ class ImageIcon extends StatelessWidget {
         child: SizedBox(width: iconSize, height: iconSize)
       );
 
-    final double iconOpacity = iconTheme.opacity;
-    Color iconColor = color ?? iconTheme.color;
+    double iconOpacity = iconTheme.opacity;
+    Color iconColor;
 
-    if (iconOpacity != null && iconOpacity != 1.0)
-      iconColor = iconColor.withOpacity(iconColor.opacity * iconOpacity);
+    if (shouldRecolor) {
+      iconColor = color ?? iconTheme.color;
+      if (iconOpacity != null && iconOpacity != 1.0) {
+        iconOpacity = iconColor.opacity * iconOpacity;
+      }
+    }
 
     return Semantics(
       label: semanticLabel,
-      child: Image(
-        image: image,
-        width: iconSize,
-        height: iconSize,
-        color: iconColor,
-        fit: BoxFit.scaleDown,
-        alignment: Alignment.center,
-        excludeFromSemantics: true,
+      child: Opacity(
```

Comment:
we definitely shouldn't use an Opacity to change the opacity of an image. See the docs for Opacity for an alternative way to do it.